### ISSUE TITLE
implement threaded queries and improve performance

### ISSUE THREAD

**Issue by mispp** [mispp/goat#2]
performance is not satisfactory on bigger result sets.

e.g.  following query kills the GUI (1500x1500 rows):

``` sql
select
   i.*
from
    information_schema.columns i
    ,information_schema.columns p
```

**Comment by mispp** [comment]
performance issue is coming from not using forward-only queries. this should be addressed in #23 but still, threaded queries should be implemented. this will also server as a basis for data export which has to happen in it's own thread.

**Comment by asw-dev** [comment]
I suspect this is an issue of using the GUI/main thread (the example query took 1.96 mins vs 1.89 in pqsl w/o outputing to the terminal). You can confirm that by running `SELECT pg_sleep(10);` and seeing it lock up the gui the same way.

I'm currently toying with a change that will fix #17 and make it much easier to implement threading safely.

**Comment by mispp** [comment]
Forgot to write 'not only'. Of course, problem is with execution of query within GUI thread, but also forward-only should improve performance significantly based on stackoverflow 

Anyhow, since example query is not complicated nor result set is that huge, GUI locking is just first reason for multi threading. 

There is a comment in #23 with links to threads which explain implementation concerns/ideas

**Comment by asw-dev** [comment]
Didn't mean to close this issue.

**Comment by mispp** [comment]
Played a bit with multithreading and implemented first version in commits 1979495b8b90057cab75f6c9a22f967ddbab852e and a59cb72e966f2e403132b020e215c1921ad66508

Query mentioned in issue description might not be a good test scenario so i tried this:

`create table a as select * from information_Schema.columns;`

`create table b as select a1.* from a a1, a a2;`

and in app i tried the following:

`select * from b`

The issue still remains that Qt will load ALL rows into memory and then display them (because of the use of model) which takes a while. 
If results sets are in milions of rows, this can take a while and #23 should address this problem.

Otherwise, this doesnt block GUI anymore.

Issue should be still open since i havent hooked up buttons and stuff to reflect query execution state, so UI improvements are next to be done here.

**Comment by mispp** [comment]
Two tabs, with the same connection running different queries causes application to crash.

Db connection cloning is needed with this multithreading. opened #29
